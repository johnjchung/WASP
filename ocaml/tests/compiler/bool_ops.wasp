Endpoint boolOps(GET:) {
	bool a = true and true;
	bool b = true and false;
	bool c = false and true;
	bool d = false and false;

	bool e = true or true;
	bool f = true or false;
	bool g = false or true;
	bool h = false or false;

	bool i = true xor true;
	bool j = true xor false;
	bool k = false xor true;
	bool l = false xor false;
}
