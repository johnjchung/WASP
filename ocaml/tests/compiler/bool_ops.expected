package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type boolOps_struct struct {
	A bool `db:"a" json:"a"`
	B bool `db:"b" json:"b"`
	C bool `db:"c" json:"c"`
	D bool `db:"d" json:"d"`
	E bool `db:"e" json:"e"`
	F bool `db:"f" json:"f"`
	G bool `db:"g" json:"g"`
	H bool `db:"h" json:"h"`
	I bool `db:"i" json:"i"`
	J bool `db:"j" json:"j"`
	K bool `db:"k" json:"k"`
	L bool `db:"l" json:"l"`
}

func main() {
	router := gin.Default()
	router.GET("/boolOps", Get_boolOps)
	router.Run(":"+os.Getenv("PORT"))
}

func boolOps_struct_wrapper() boolOps_struct {
	return boolOps_struct{A: bool(true) && bool(true), B: bool(true) && bool(false), C: bool(false) && bool(true), D: bool(false) && bool(false), E: bool(true) || bool(true), F: bool(true) || bool(false), G: bool(false) || bool(true), H: bool(false) || bool(false), I: bool(true) != bool(true), J: bool(true) != bool(false), K: bool(false) != bool(true), L: bool(false) != bool(false)}
}

func Get_boolOps(c *gin.Context) {
	boolOps_struct := boolOps_struct_wrapper()
	c.JSON(200, boolOps_struct)
}
